addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -332,11 +350,88 @@ private ApplicationMatch addIfMatches(
+            LOGGER.debug("{}: {}", loc, evidence);
+        }
+    }
+
+    private void raiseAlert(HttpMessage msg, ApplicationMatch appMatch) {
+        if (raiseAlerts) {
+            LOGGER.debug(
+                    "Adding \"{}\" alert for \"{}\"",
+                    appMatch.getApplication().getName(),
+                    msg.getRequestHeader().getURI());
+            ExtensionAlert extAlert =
+                    Control.getSingleton().getExtensionLoader().getExtension(ExtensionAlert.class);
+            if (extAlert != null) {
+                Alert alert = createAlert(msg, appMatch);
+                extAlert.alertFound(alert, msg.getHistoryRef());
+            }
+        }
+    }
+
+    Alert createAlert(HttpMessage msg, ApplicationMatch appMatch) {
+        Application app = appMatch.getApplication();
+
+        Builder builder = Alert.builder();
+        builder.setPluginId(PLUGIN_ID)
+                .setName(Constant.messages.getString("wappalyzer.alert.name.prefix", app.getName()))
+                .setMessage(msg)
+                .setRisk(Alert.RISK_INFO)
+                .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                .setUri(msg.getRequestHeader().getURI().toString())
+                .setDescription(getDesc(app))
+                .setCweId(200)
+                .setWascId(13);
+        if (!appMatch.getEvidences().isEmpty()) {
+            builder.setEvidence(appMatch.getEvidences().stream().findFirst().get());
+        }
+        if (app.getCpe() != null && !app.getCpe().isBlank()) {
+            builder.setOtherInfo(
+                    Constant.messages.getString("wappalyzer.alert.otherinfo.cpe", app.getCpe()));
+        }
+        if (appMatch.getVersion() != null && !appMatch.getVersions().isEmpty()) {
+            builder.setOtherInfo(getVersionInfo(appMatch, builder));
+        }
+        if (app.getWebsite() != null && !app.getWebsite().isBlank()) {
+            builder.setReference(app.getWebsite());
+        }
+        return builder.build();
+    }
+
+    private static String getDesc(Application app) {
+        String desc =
+                Constant.messages.getString(
+                        "wappalyzer.alert.desc",
+                        collectionToString(app.getCategories()),
+                        app.getName());
+        if (app.getDescription() != null && !app.getDescription().isEmpty()) {
+            desc =
+                    desc
+                            + Constant.messages.getString(
+                                    "wappalyzer.alert.desc.extended", app.getDescription());
+        }
+        return desc;
+    }
+
+    private static String getVersionInfo(ApplicationMatch appMatch, Builder builder) {
+        String versionInfo =
+                Constant.messages.getString(
+                        "wappalyzer.alert.otherinfo.version",
+                        collectionToString(appMatch.getVersions()));
+        Alert alert = builder.build();
```

Comment:
This is odd, pass the info as String instead of Builder.

Reply:
How do I get the optional existing other info without checking the alert?

Reply:
I guess I could introduce a boolean or pass both optional strings at the same time.

Reply:
Done